static/src/javascripts/projects/common/modules/commercial/adblock-messages.js

```diff
@@ -9,18 +9,18 @@ define([
     storage,
     userFeatures
 ) {
-    function sharedRules() {
-        var alreadyVisted = storage.local.get('gu.alreadyVisited') || 0;
-
-        return detect.getBreakpoint() !== 'mobile' && detect.adblockInUse() && config.switches.adblock && alreadyVisted > 1;
-    }
-
     function noAdblockMsg() {
-        return sharedRules() && userFeatures.isPayingMember();
+        var alreadyVisited = storage.local.get('gu.alreadyVisited') || 0;
+
+        return detect.adblockInUse() && detect.getBreakpoint() !== 'mobile' && (
+                alreadyVisited <= 1 ||
+                !config.switches.adblock ||
+                (config.switches.adblock && alreadyVisited > 1 && userFeatures.isPayingMember()));
     }
 
     function showAdblockMsg() {
-        return sharedRules() && !userFeatures.isPayingMember();
+        var alreadyVisited = storage.local.get('gu.alreadyVisited') || 0;
+        return detect.getBreakpoint() !== 'mobile' && detect.adblockInUse() && config.switches.adblock && alreadyVisited > 1 && !userFeatures.isPayingMember();
```

Comment:
This line is a bit long. Also, I would put less computationally heavy switches at the start, so they can short-circuit the evaluation.